### PR TITLE
Make HTTP progress visible at the information level

### PR DESCRIPTION
**Pull request by joelverhagen**
Per @davidfowl's request, show the HTTP operations during the restore at the information (default) logging level.

@emgarten @anurse

### CODE REVIEW

**Comment by analogrelay** [comment]
We might actually want to keep these messages as `Verbose` and just make verbose the default and have a `--quiet` setting to set the default back to `Info`. We do still want a way to suppress the HTTP progress but keep the other status messages.

**Comment by joelverhagen** [comment]
@anurse, sounds good. I'll talked to @emgarten about this.

**Comment by analogrelay** [comment]
FYI: The specific switch we use to enable a quiet-mode doesn't really worry me. Keeping `--verbosity` is fine, as long as there is a way to make it quieter :)

**Comment by joelverhagen** [comment]
Spoke to @emgarten offline about this. Added an additional log level called "minimal", which is between information and warning. The default log is information level. I move all currently information logs to minimal and moved the HTTP status logs to information.

The equivalent to DNU `--quiet` is `--verbosity minimal`, which no longer shows the HTTP progress.

There is currently a hack in the Visual Studio `OnBuildPackageRestorer` progress popup code that ignores information (and now minimal) level logs if they start with a space. This should be cleaned up so that logs starting with a space are logged as information and only minimal is observed by the VS code. This will eliminate the need for this magic.

**Comment by davidfowl** [comment]
Screenshot?

File: src/NuGet.Core/NuGet.Protocol.Core.v3/RemoteRepositories/RemoteV2FindPackageByIdResource.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ into nextLink
                 }
                 catch (Exception ex) when (retry < 2)
                 {
-                    Logger.LogInformation(string.Format("Warning: FindPackagesById: {1}\r\n  {0}", ex.Message, id));
+                    Logger.LogMinimal(string.Format("Warning: FindPackagesById: {1}\r\n  {0}", ex.Message, id));
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/RemoteRepositories/RemoteV2FindPackageByIdResource.cs]: It seems like these should be at the warning level

File: src/NuGet.Core/NuGet.Commands/RestoreCommand.cs [hunk 2]
```diff
@@ -80,7 +80,7 @@ public Task<RestoreResult> ExecuteAsync()
                 // The lock file was locked, but the project.json is out of date
                 relockFile = true;
                 _request.ExistingLockFile.IsLocked = false;
-                _logger.LogInformation(Strings.Log_LockFileOutOfDate);
+                _logger.LogMinimal(Strings.Log_LockFileOutOfDate);
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand.cs]: This should be a warning

File: src/NuGet.Core/NuGet.CommandLine.XPlat/CommandOutputLogger.cs [hunk 3]
```diff
@@ -79,18 +85,21 @@ private void LogInternal(LogLevel logLevel, string message)
...
                         caption = "debug";
                         break;
+                    case LogLevel.Verbose:
+                        caption = "trace";
+                        break;
                     case LogLevel.Information:
                         caption = "info ";
                         break;
+                    case LogLevel.Minimal:
+                        caption = "mimim";
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/CommandOutputLogger.cs]: what if we just called this `"log  "` or `"output"` ?

**Comment by joelverhagen** [comment]
Examples:
- [Default logging when restoring `dotnet new`](https://gist.github.com/joelverhagen/c3c1dc4d0479040faea7/d8de3c32ff20a7a8e5aafb44fa7b11f8e5d32a69)
- [`--verbosity minimal`](https://gist.github.com/joelverhagen/c3c1dc4d0479040faea7/8bb270ad98f3557611a5aa5893db3367fc2f3bd6)

File: src/NuGet.Core/NuGet.CommandLine.XPlat/CommandOutputLogger.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/CommandOutputLogger.cs]: Fixed in latest. Rebased off of latest `dev`.

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**